apis/Google.Devtools.AspNet/Google.Devtools.AspNet.IntegrationTests/CloudTrace/CloudTraceIntregrationTest.cs

```diff
@@ -0,0 +1,288 @@
+
+            Dictionary<string, string> annotation = new Dictionary<string, string>
+            {
+                { "annotation-key", "annotation-value" },
+                { "some-key", "some-value" }
+            };
+
+            tracer.StartSpan(rootSpanName);
+            Thread.Sleep(10);
+            tracer.AnnotateSpan(annotation);
+            tracer.EndSpan();
+
+            Trace trace = await GetTrace(rootSpanName);
+            Assert.NotNull(trace);
+            Assert.Single(trace.Spans);
+            Assert.True(Tests.TraceUtils.IsValidAnnotation(trace.Spans[0], annotation));
+        }
+
+        [Fact]
+        public async Task Trace_SimpleStacktrace()
+        {
+            string rootSpanName = CreateRootSpanName(nameof(Trace_SimpleStacktrace));
+            GrpcTraceConsumer consumer = CreateGrpcTraceConsumer();
+            SimpleManagedTracer tracer = CreateSimpleManagedTracer(consumer);
+
+            tracer.StartSpan(rootSpanName);
+            Thread.Sleep(10);
+            tracer.SetStackTrace(new StackTrace(true));
+            tracer.EndSpan();
+
+            Trace trace = await GetTrace(rootSpanName);
+            Assert.NotNull(trace);
+            Assert.Single(trace.Spans);
+
+            MapField<string, string> labels = trace.Spans[0].Labels;
+            Assert.True(labels.ContainsKey(Labels.StackTrace));
+            Assert.Contains(nameof(Trace_SimpleStacktrace), labels[Labels.StackTrace]);
+            Assert.Contains(nameof(CloudTraceIntregrationTest), labels[Labels.StackTrace]);
+        }
+
+        [Fact]
+        public async Task Trace_MultipleSpans()
+        {
+            string rootSpanName = CreateRootSpanName(nameof(Trace_MultipleSpans));
+            GrpcTraceConsumer consumer = CreateGrpcTraceConsumer();
+            SimpleManagedTracer tracer = CreateSimpleManagedTracer(consumer);
+
+            Dictionary<string, string> annotation = new Dictionary<string, string>
+            {
+                { "annotation-key", "annotation-value" }
+            };
+
+            tracer.StartSpan(rootSpanName);
+            Thread.Sleep(10);
+            tracer.StartSpan("child-one");
+            tracer.SetStackTrace(new StackTrace(true));
+            Thread.Sleep(20);
+            tracer.EndSpan();
+            tracer.StartSpan("child-two");
+            Thread.Sleep(5);
+            tracer.StartSpan("grandchild-one", StartSpanOptions.Create(SpanKind.RpcClient));
+            Thread.Sleep(50);
+            tracer.EndSpan();
+            tracer.StartSpan("grandchild-two");
+            Thread.Sleep(10);
+            tracer.AnnotateSpan(annotation);
+            tracer.EndSpan();
+            tracer.EndSpan();
+            tracer.EndSpan();
+
+            Trace trace = await GetTrace(rootSpanName);
+            Assert.NotNull(trace);
+            Assert.Equal(5, trace.Spans.Count);
+
+            TraceSpan root = trace.Spans.Where(s => s.Name.Equals(rootSpanName)).First();
```

Comment:
All of these `Where(condition).First()` can be simplified to `First(condition)`.